mne/viz/topomap.py

```diff
@@ -1502,6 +1499,14 @@ def plot_evoked_topomap(evoked, times="auto", ch_type=None, layout=None,
         .. versionadded:: 0.18
     %(topomap_sphere_auto)s
     %(topomap_border)s
+    nrows : int | 'auto'
+        The number of rows of topographies to plot. Defaults to 1. If 'auto',
+        obtains the number of rows depending on the amount of times to plot
+        and the number of cols. Not valid when times == 'interactive'.
+    ncols : int | 'auto'
+        The number of columns of topographies to plot. If 'auto' (default),
+        obtains the number of columns depending on the amount of times to plot
+        and the number of rows. Not valid when times == 'interactive'.
```

Comment:
What happens if both are auto (needs to be documented)? If you haven't already decided, I would say in this case we could have `ncols` set to `min(20, len(times))` and `nrow` set to whatever it needs to be to plot everything.

Also this is missing `.. versionadded:: 0.20` directives

Reply:
> What happens if both are auto (needs to be documented)? If you haven't already decided, I would say in this case we could have `ncols` set to `min(20, len(times))` and `nrow` set to whatever it needs to be to plot everything.

@larsoner Good point. Currently the behaviour is to raise an error. `plot_projs_topomap` has another behaviour (but does not expose the `nrows` and `ncols` parameters):
```Python
nrows = math.floor(math.sqrt(n_projs))
ncols = math.ceil(n_projs / nrows)
```

Shall I add `ncols = min(20, len(n_cells))` or use the same behaviour as `plot_projs_topomap`?

> Also this is missing `.. versionadded:: 0.20` directives
Just added it.

Reply:
I guess unifying with `plot_projs_topomap` (making it square) would be the best option. The difference would be that `plot_projs_topomap` would have `n_rows='auto'` and `evoked.plot_topomap` would have `n_rows=1` by default. The justification would be that it's a lot easier to accidentally plot hundreds of evoked topomaps and kill your machine than it is to do it with projectors (which are typically very limited in number).

Reply:
I agree that it's easy to accidentally ask for too many plots. If the plot is not multiline and more than 20 timepoints are asked, then `_process_times` raises an error.

So in the case of `nrows=1` then the plot is not multiline and `_process_times` will raise the error. If you specify at least nrows=2, then you can still have 2 rows of 100 of plots.

The other way would be to restrict the number of columns to 20, which will also allow you to plot hundreds of topomaps. We can also restrict the number of rows to 20, giving a maximum of 400 topomaps in the worst case scenario.

Or we can do as matplotlib does, where more than 20 open figures raises a warning.

I can implement any of these 3 options. Not sure which one is the best.

Reply:
> The other way would be to restrict the number of columns to 20, which will also allow you to plot hundreds of topomaps. We can also restrict the number of rows to 20, giving a maximum of 400 topomaps in the worst case scenario.

Sounds reasonable to me

Reply:
I don't like restricting the number of columns to 20.  What if someone wants to generate a landscape figure that is 25 columns x 10 rows, for example?  During exploratory analysis, or for a supplementary figure, this doesn't seem wildly unreasonable to me.

I think I prefer this: 

- `nrows=1, ncols='auto'` is the default
- `nrows='auto', ncols='auto'`: do the `sqrt` thing
- number of times > 25: warn, don't error (25 is the limit for the ICA topomap function to trigger a second, separate figure, that's why I picked 25)
- don't put any upper limit on the specific numbers passed to `nrows` or `ncols`